tests/components/energy/test_sensor.py

```diff
@@ -31,6 +32,17 @@
 from tests.components.recorder.common import async_wait_recording_done
 
 
+@pytest.fixture(autouse=True)
+def garbage_collection():
+    """Make sure garbage collection is run between all tests.
+
+    There are unknown issues with GC triggering during a test
+    case, leading to the test breaking down. Make sure we
+    clean up between each testcase to avoid this issue.
+    """
+    gc.collect()
```

Comment:
Shouldn't this be 
```suggestion
    yield
    gc.collect()
```
so it runs at the end of a test instead of the start ?

Reply:
Indeed. Will fix.